Import Pattern Accessibility documentation

chloepoulterdesign opened the pull request:
There isn't an issue for this, however I am acting on the output of Fang Shih's audit of pattern accessibility. You can see her findings [here](https://ibm.box.com/s/m7bh9pul6p57fgedjse5193rxyn66ujp) (w3 log-in required).

#### Changelog

**New**

- Section called accessibility
- Anchor link 
- Modal & Single file sub-section stepping through triggering options and keyboard interaction orders
- Full page & side panel sub-section covering keyboard interaction orders

jeanservaas reviewed (changes requested):
Hi @chloepoulterdesign thanks!

Some of what I commented on the export pattern will have the same impact here:

![image](https://user-images.githubusercontent.com/43144260/99717748-72c23a00-2a6f-11eb-8c4d-341f9ec4a26c.png)

I'm going to attach those templates that I built for full screen UI here too so that we're all using the same aspect ratios, margins (depending on the aspect ratio) and background browser asset.

Please make sure these images are using either a 16:9 or a 3:2 aspect ratio... a little taller image might be nice because the bottom / top margin is a little tight. I think that add file button can also be attached to the input field in these cases.

![image](https://user-images.githubusercontent.com/43144260/99718009-cf255980-2a6f-11eb-9188-070121a399a9.png)

Would you get that native OS red number for dragged files in product? Also, I know Mac on desktop has the red number but the round tags around the file names do not look like native mac.

![image](https://user-images.githubusercontent.com/43144260/99718132-0bf15080-2a70-11eb-9b3a-5230ded8c986.png)

We don't use drawn arrows in the design language. I think you can make your point about tabbing with text only, either above the image or below in the caption

In dropdown, we talk about key commands and show similar sequences, there might be some good inspiration here:

https://www.carbondesignsystem.com/components/dropdown/usage

![image](https://user-images.githubusercontent.com/43144260/99719052-6e971c00-2a71-11eb-8bb1-e5a98fb85ca8.png)

In this section, I might actually keep the side panels examples and guidance together under one sub head and the full page guidance together under another subhead so the user's not jumping around 

![image](https://user-images.githubusercontent.com/43144260/99719656-380dd100-2a72-11eb-8ad4-7fb07ccd3ab1.png)

Also, I think some of the 608 usage full screen templates will work for you here and give you more margin / eliminate that shadow.

![image](https://user-images.githubusercontent.com/43144260/99719761-5d9ada80-2a72-11eb-9059-36397da62b34.png)

Same with Export, I'm going to give you the anatomy callout symbols that we use in a sketch file. Also I might restructure the content a little here. Anatomy images are usually only used once or twice in a pattern. A lot of the parts that you see in the initial image are also in the following images. What you're showing here seems to be more of a sequence of behaviors rather than an anatomy. 

I think we actually need better guidance around a situation like this. I'm not saying you can't do this, but I think you could edit this area down a lot so there's less redundancy. 

In Carbon we tend to isolate pieces like this to show their states (file uploader is a good usage tab to look at for what you've got going:

![image](https://user-images.githubusercontent.com/43144260/99720387-4c9e9900-2a73-11eb-96c5-0be171438c88.png)


![image](https://user-images.githubusercontent.com/43144260/99719821-70151400-2a72-11eb-86a3-62f11c637df5.png)

Also, keep all margins even in situations like this, and try not to use a margin of less than 32px all the way around

![image](https://user-images.githubusercontent.com/43144260/99720529-7c4da100-2a73-11eb-97ad-d36788586d84.png)

Remember your feedback and related sections too. File uploader for instance would be related and the Create flow too.

chloepoulterdesign commented:
On discussion with @jeanservaas I've removed the imagery in the accessibility section. 
I've tidied things up to make it clearer without images. 
I've renamed the sections to better explain that the behaviour varies depending on whether it is a `single` or `multi` file import/upload. 
I've raised #2013 to work on the image quality in the pattern documentation in future.

jeanservaas reviewed (commented):
Hey @chloepoulterdesign 

Same apology here, you're not responsible for updating images that are already on the site, haha sorry. 

As for the added accessibility content, could we treat it the same way as you did for the Export pattern added content, which I just approved (i.e. following the Carbon pattern and not trying to add all those anatomy images)?

chloepoulterdesign commented:
Hey @jeanservaas
I have made the same changes as in Export here - it looks like the Vercel deployment failed when I committed them.
I can't trigger the redeployment, could you or someone in the team have a go at that?
![image](https://user-images.githubusercontent.com/47979346/100331398-02b03880-2fc8-11eb-9438-2cf22f9d2aa2.png)

jillianhowarth commented on src/pages/community/patterns/import-pattern/index.mdx:
```diff
@@ -175,7 +175,118 @@ If your product supports importing from a URL, use this method.
+the file selection methods are in disabled state. Focus cycles between the
+selected files "remove" actions, and moving to the next section of the page.
+
+1. On returning to the page/panel from the final File uploader/"Add file"
```
```suggestion
1. On returning to the page/panel from the final file uploader/add file
```

jillianhowarth commented on src/pages/community/patterns/import-pattern/index.mdx:
```diff
@@ -175,7 +175,118 @@ If your product supports importing from a URL, use this method.
+
+1. On returning to the page/panel from the final File uploader/"Add file"
+   interaction, (after any optional loading state) focus is applied to the
+   "remove" icon on the uploaded files.
```
```suggestion
   remove icon on the uploaded files.
```

jillianhowarth commented on src/pages/community/patterns/import-pattern/index.mdx:
```diff
@@ -175,7 +175,118 @@ If your product supports importing from a URL, use this method.
+1. On returning to the page/panel from the final File uploader/"Add file"
+   interaction, (after any optional loading state) focus is applied to the
+   "remove" icon on the uploaded files.
+2. On keying `tab`, focus moves through the "remove" icon of each uploaded file.
```
```suggestion
2. On keying `Tab`, focus moves through the remove icon of each uploaded file.
```

jillianhowarth commented on src/pages/community/patterns/import-pattern/index.mdx:
```diff
@@ -175,7 +175,118 @@ If your product supports importing from a URL, use this method.
+   interaction, (after any optional loading state) focus is applied to the
+   "remove" icon on the uploaded files.
+2. On keying `tab`, focus moves through the "remove" icon of each uploaded file.
+3. On continued use of `tab`, focus is applied to the next section or
```
```suggestion
3. On continued use of `Tab`, focus is applied to the next section or
```

jillianhowarth commented on src/pages/community/patterns/import-pattern/index.mdx:
```diff
@@ -175,7 +175,118 @@ If your product supports importing from a URL, use this method.
+3. On continued use of `tab`, focus is applied to the next section or
+   interaction in the page.
+
+## Related
```
```suggestion
```

jillianhowarth reviewed (changes requested):
@chloepoulterdesign this is looking great! Just some minor capitalization/punctuation fixes 🙂

chloepoulterdesign commented on src/pages/community/patterns/import-pattern/index.mdx:
```diff
@@ -175,7 +175,118 @@ If your product supports importing from a URL, use this method.
+   launched from an overflow menu, the user may navigate to Import within the
+   overflow menu using the up/down arrow keys.
+2. The focus state is applied to the button/menu item.
+3. The user clicks “enter” to launch the Import modal.
```
Should this be `Return` rather than `Enter`?

jeanservaas reviewed (approved):
lgtm thanks Chloe!!

jillianhowarth reviewed (approved):
Great work!! 🙂
